#! /usr/bin/env python

# Declare parameters that control the heatmap builder

PACKAGE='opt_utils'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#########################
## Heatmap parameters ##
#########################
# Max size of the heatmap (in meters):
gen.add("map_real_size", double_t, 0, "Max size of the heatmap (in meters)", 20.0, 0.0, 100.0)
# Max size of the heatmap (in pixels):
gen.add("map_pixel_size", int_t, 0, "Max size of the heatmap (in pixels)", 400, 100, 400)
# Heatmap publishing frame rate:
gen.add("rate", int_t, 0, "Heatmap publishing frame rate", 30, 10, 300) 
# Flag enabling heatmap reset:
gen.add("reset_heatmap", bool_t, 0, "Flag enabling heatmap reset", False)

# First string value is node name, used only for generating documentation
# Second string value ("HeatmapBuilder") is name of class and generated
#    .h file, with "Config" added, so class HeatmapBuilderConfig
exit(gen.generate(PACKAGE, "heatmap_builder", "HeatmapBuilder"))
